src/modules/Content/index.tsx

```diff
@@ -212,31 +216,50 @@ function useSearchState(contentQuery: QueryResult<ContentQuery, ContentQueryVari
      * depending on the search matches
      */
     const body = contentQuery.state.data?.content[0]?.body;
-    const bodyWithMatches: ContentBody[] = useMemo(
-        () =>
-            state.searchTerm && body
-                ? findAll({
-                      autoEscape: true,
-                      caseSensitive: false,
-                      searchWords: [state.searchTerm],
-                      textToHighlight: body,
-                  }).map(({ highlight, start, end }, index) => ({
-                      highlight,
-                      id: `content-body-search-term-chunk-${index}`,
-                      text: body.substr(start, end - start),
-                  }))
-                : [
-                      {
-                          highlight: false,
-                          id: 'content-body',
-                          text: body || '',
-                      },
-                  ],
-        [body, state.searchTerm]
-    );
+    const bodyWithMatches: ContentBody[][] | null = useMemo(() => {
+        if (body) {
+            const nodes: NodeListOf<ChildNode> = new DOMParser().parseFromString(body, 'text/html').body.childNodes;
```

Comment:
Hopefully this is ever a perf issue. For filings it could be with how large they are. Did you try passing a custom regex that ignores text between `<>`?

Reply:
Yeah that was my first approach, but the issue was when I overloaded their `defaultFindChunks` function and ignored any matches inside `<>`, I would either lose the paragraphs (if I stripped the html first), or if I preserved the tags (and just skipped them when they matched), it would mess up the indexing. Even when I tried to fix the indexing, the html that got rendered at the end (using `dangerouslySetInnerHTML`) would get messed up.

so searching for `w` in `<p>Hello world</p>` could result in

```
Hello

world
```

Reply:
Happy to revert and screenshare if you want to see what I mean.